### PR TITLE
Extension for projects, folders, items

### PR DESCRIPTION
**Pull request by camillaol**
Functionality extended to include ability to:
* List all projects for a given hub
* Get project details for a given project
* List all folders for a given project
* Get folder contents for a given folder
* Get item details for a given item

### CODE REVIEW

File: dm/projects.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	path := api.Host + api.HubAPIPath
+
+	return getProjectDetails(path, hubKey, projectKey, bearer.AccessToken)
+}
+
+
+/*
+ *	SUPPORT FUNCTIONS
+ */
+func listProjects(path, hubKey, id, extension, page, limit string, token string) (result HubDetails, err error) {
```
Comment (monicatie) [inline dm/projects.go]: Might be great to break out the query params. I liked how Contentul's go sdk did it (as an example)

https://github.com/contentful-labs/contentful-go/blob/master/query.go

File: dm/projects.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	err = decoder.Decode(&result)
+
+	return
+}
+
+func getProjectDetails(path, hubKey, projectKey, token string) (result ProjectDetails, err error) {
+	task := http.Client{}
+
+	req, err := http.NewRequest("GET",
```
Comment (monicatie) [inline dm/projects.go:96]: [Nice to have] Saw this great request debugging feature in https://github.com/contentful-labs/contentful-go/blob/d6275d6dde296c841829bedce9fb007114af85c1/contentful.go#L169

It outputs what the curl would be!

File: dm/folders.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+// NewFolderAPIWithCredentials returns a Folder API client with default configurations
+func NewFolderAPIWithCredentials(ClientID string, ClientSecret string) FolderAPI {
+	return FolderAPI{
+		oauth.NewTwoLeggedClient(ClientID, ClientSecret),
+		"/data/v1/projects",
+	}
+}
+
+// ListBuckets returns a list of all buckets created or associated with Forge secrets used for token creation
+func (api FolderAPI) ListFolders(projectKey, folderKey string) (result FolderDetails, err error) {
```
Comment (monicatie) [inline dm/folders.go]: GetFolderDetails? This one appears to get a specific folder

File: dm/projects.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	bearer, err := api.Authenticate("data:read")
+	if err != nil {
+		return
+	}
+
+	path := api.Host + api.HubAPIPath
+
+	return listProjects(path, hubKey, "", "", "", "", bearer.AccessToken)
+}
+
```
Comment (monicatie) [inline dm/projects.go:22]: Note: missing topFolders functionality
https://forge.autodesk.com/en/docs/data/v2/reference/http/hubs-hub_id-projects-project_id-topFolders-GET/

File: dm/items.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
+package dm
```
Comment (monicatie) [inline dm/items.go:1]: Lets add the `tip` and `versions` too~

**Review by monicatie — CHANGES_REQUESTED** [review]

File: dm/folders.go (hunk 3 above)
Comment (camillaol) [reply dm/folders.go]: updated!

File: dm/projects.go (hunk 4 above)
Comment (camillaol) [reply dm/projects.go:22]: done!

File: dm/items.go (hunk 5 above)
Comment (camillaol) [reply dm/items.go:1]: done!

File: dm/projects.go (hunk 2 above)
Comment (camillaol) [reply dm/projects.go:96]: per chat, punting! captured in trello ticket

File: dm/projects.go (hunk 1 above)
Comment (camillaol) [reply dm/projects.go]: per chat, punting! captured in trello ticket

**Review by camillaol — COMMENTED** [review]
Updated for requested features and bug fixes. Punting on the nice-to-haves (cleaner query string param handling, and request debugger), which are captured in Trello tickets .

**Pull request merged**